hedera-mirror-rosetta/app/services/block/block_service_test.go

```diff
@@ -0,0 +1,233 @@
+		ParentHash:          "parenthash",
+	}
+}
+
+func exampleBlockRequest() *rTypes.BlockRequest {
+	index := int64(100)
+	hash := "somehashh"
+
+	return &rTypes.BlockRequest{
+		NetworkIdentifier: &rTypes.NetworkIdentifier{
+			Blockchain:           "Hedera",
+			Network:              "hhh",
+			SubNetworkIdentifier: nil,
+		},
+		BlockIdentifier: &rTypes.PartialBlockIdentifier{
+			Index: &index,
+			Hash:  &hash,
+		},
+	}
+}
+
+func exampleBlockResponse() *rTypes.BlockResponse {
+	return &rTypes.BlockResponse{
+		Block: &rTypes.Block{
+			BlockIdentifier: &rTypes.BlockIdentifier{
+				Index: 1,
+				Hash:  "0x123jsjs",
+			},
+			ParentBlockIdentifier: &rTypes.BlockIdentifier{
+				Index: 2,
+				Hash:  "0xparenthash",
+			},
+			Timestamp: 1,
+			Transactions: []*rTypes.Transaction{
+				{
+					TransactionIdentifier: &rTypes.TransactionIdentifier{Hash: "123"},
+					Operations:            []*rTypes.Operation{},
+					Metadata:              nil,
+				},
+				{
+					TransactionIdentifier: &rTypes.TransactionIdentifier{Hash: "246"},
+					Operations:            []*rTypes.Operation{},
+					Metadata:              nil,
+				},
+			},
+			Metadata: nil,
+		},
+		OtherTransactions: nil,
+	}
+}
+
+func dummyTransaction(hash string) *types.Transaction {
+	return &types.Transaction{
+		Hash:       hash,
+		Operations: nil,
+	}
+}
+
+func transactionRequest() *rTypes.BlockTransactionRequest {
+	return &rTypes.BlockTransactionRequest{
+		NetworkIdentifier: &rTypes.NetworkIdentifier{
+			Blockchain: "someblockchain",
+			Network:    "somenetwork",
+			SubNetworkIdentifier: &rTypes.SubNetworkIdentifier{
+				Network:  "somesubnetwork",
+				Metadata: nil,
+			},
+		},
+		BlockIdentifier: &rTypes.BlockIdentifier{
+			Index: 1,
+			Hash:  "someblockhash",
+		},
+		TransactionIdentifier: &rTypes.TransactionIdentifier{Hash: "somehash"},
+	}
+}
+
+func TestNewBlockAPIService(t *testing.T) {
+	repository.Setup()
+	baseService := base.NewBaseService(repository.MBlockRepository, repository.MTransactionRepository)
+	blockService := NewBlockAPIService(baseService)
+
+	assert.IsType(t, &BlockAPIService{}, blockService)
+}
+
+func TestBlock(t *testing.T) {
+	// given:
+	exampleTransactions := []*types.Transaction{
+		dummyTransaction("123"),
+		dummyTransaction("246"),
+	}
+
+	repository.Setup()
+	repository.MBlockRepository.On("FindByIdentifier").Return(block(), repository.NilError)
+	repository.MTransactionRepository.On("FindBetween").Return(exampleTransactions, repository.NilError)
+
+	// when:
+	res, e := getSubject().Block(nil, exampleBlockRequest())
+
+	// then:
+	assert.Nil(t, e)
+	assert.Equal(t, exampleBlockResponse(), res)
+}
+
+func TestBlockThrowsWhenFindByIdentifierFails(t *testing.T) {
+	// given:
+	repository.Setup()
+	repository.MBlockRepository.On("FindByIdentifier").Return(
+		repository.NilBlock,
+		&rTypes.Error{},
+	)
+
+	// when:
+	res, e := getSubject().Block(nil, exampleBlockRequest())
+
+	// then:
+	assert.NotNil(t, e)
```

Comment:
I think that everytime where you check that the error is of type Error you are implicitly checking that error is not nil, meaning that if you have:
```Go
assert.IsType(t, &rTypes.Error{}, e)
```
you have a implicit check:

```Go
assert.NotNil(t, e)
```